Sources/AsyncKit/ConnectionPool/EventLoopConnectionPool.swift

```diff
@@ -55,16 +60,20 @@ public final class EventLoopConnectionPool<Source> where Source: ConnectionPoolS
     ///     - source: Creates new connections when needed.
     ///     - maxConnections: Limits the number of connections that can be open.
     ///                       Defaults to 1.
+    ///     - requestTimeout: Timeout for requesting a new connection.
+    ///                       Defaults to 10 seconds.
     ///     - logger: For lifecycle logs.
     ///     - on: Event loop.
     public init(
         source: Source,
         maxConnections: Int,
+        requestTimeout: TimeAmount = .seconds(10),
```

Comment:
FWIW, Redis's connection pool is using a timeout of 60 seconds: https://gitlab.com/Mordil/RediStack/-/merge_requests/116/diffs#8fde653c2095c02534d7072fa60cdb002ac223c5_0_208

The docs note: 
> for maximum compatibility with the existing implementation, we use a fairly-long timeout:

cc @lukasa

Reply:
Yeah, we were very defensive there. Maybe more than we should have been.